sdk/core/core-lro/src/poller/models.ts

```diff
@@ -173,12 +181,28 @@ export type CancelOnProgress = () => void;
 /**
  * A simple poller interface.
  */
-export interface SimplePollerLike<TState extends OperationState<TResult>, TResult> {
+export interface PollerLike<TState extends OperationState<TResult>, TResult>
+  extends Promise<TResult> {
+  /**
+   * Returns true if the poller has finished polling.
+   */
+  readonly isDone: boolean;
+  /**
+   * Returns the state of the operation.
+   */
+  readonly operationState: TState | undefined;
+  /**
+   * Returns the result value of the operation,
+   * regardless of the state of the poller.
+   * It can return undefined or an incomplete form of the final TResult value
+   * depending on the implementation.
```

Comment:
```suggestion
   * The result value of the operation, regardless of the state of the poller.
   * It can be undefined or an incomplete form of the final TResult value
   * depending on the implementation.
```